Configuration/Skimming/src/LeptonSkimming.cc

```diff
@@ -0,0 +1,487 @@
+	 tempTracks.clear(); 
+         tempTracks.push_back(*tranobj); tempTracks.push_back(*trantrk2);
+         LLvertex = theKalmanFitter.vertex(tempTracks);
+         if (!LLvertex.isValid()) continue;
+	 if (ChiSquaredProbability(LLvertex.totalChiSquared(),LLvertex.degreesOfFreedom())<Probee_Cut)  continue;
+         if (ZvertexTrg>-1000000 && fabs(ZvertexTrg-LLvertex.position().z())>EpairZvtx_Cut ) continue;
+	 GlobalError err =LLvertex.positionError();
+	 GlobalPoint Dispbeamspot(-1*((theBeamSpot->x0()-LLvertex.position().x())+(LLvertex.position().z()-theBeamSpot->z0()) * theBeamSpot->dxdz()),-1*((theBeamSpot->y0()-LLvertex.position().y())+ (LLvertex.position().z()-theBeamSpot->z0()) * theBeamSpot->dydz()), 0);
+          math::XYZVector pperp((vel1+vel2).Px(),(vel1+vel2).Py(),0);
+          math::XYZVector vperp(Dispbeamspot.x(),Dispbeamspot.y(),0.);
+          float tempCos=vperp.Dot(pperp)/(vperp.R()*pperp.R());
+          if (tempCos<Cosee_Cut) continue;
+	  cleanedObjTracks.push_back(obj);
+          cleanedPairTracks.push_back(trk2);
+	  Epair_ObjectIndex.push_back(object_container.at(iobj));
+          Epair_ObjectId.push_back(object_id.at(iobj));
+          Epair_TrkIndex.push_back(Trk_container.at(itrk2));
+            
+	}
+    }
+//    cout<<trk_index<<" comb "<<Epair_ObjectId.size()<<endl;   
+    
+    // B recontrsuvtion
+     TLorentzVector vK; 
+    for(unsigned int iobj=0; iobj<cleanedObjTracks.size(); iobj++){
+      auto objtrk=cleanedObjTracks.at(iobj);
+      auto pairtrk=cleanedPairTracks.at(iobj);
+      auto tranobj=std::make_shared<reco::TransientTrack>(reco::TransientTrack(*objtrk,&(*bFieldHandle)));
+      auto tranpair=std::make_shared<reco::TransientTrack>(reco::TransientTrack(*pairtrk,&(*bFieldHandle)));
+      unsigned int index=Epair_ObjectIndex.at(iobj);
+      if ( Epair_ObjectId.at(iobj)==13) vel1.SetPtEtaPhiM(muon_pt.at(index),muon_eta.at(index),muon_phi.at(index),0.0005);
+      
+      else vel1.SetPtEtaPhiM(el_pt.at(index),el_eta.at(index),el_phi.at(index),0.0005);
+       
+      for(auto & trk: cleanedTracks){
+	//        unsigned int itrk= &trk-&cleanedTracks[0];
+	//auto trk=cleanedTracks.at(itrk);
+         if(deltaR(objtrk->eta(),objtrk->phi(),trk->eta(),trk->phi())<TrkObjExclusionCone) continue;
+	 if (trk->pt()<PtKTrack_Cut) continue;
+         if (fabs(trk->dxy(vertex_point))/trk->dxyError()<Ksdxy_Cut) continue;
+         if (trk->charge()==pairtrk->charge() && deltaR(pairtrk->eta(),pairtrk->phi(),trk->eta(),trk->phi())<TrkTrkMinDR_Cut) continue;
+         
+         vel2.SetPtEtaPhiM(pairtrk->pt(),pairtrk->eta(),pairtrk->phi(),0.0005);
+         vK.SetPtEtaPhiM(trk->pt(),trk->eta(),trk->phi(),0.493);
+         if ((vel1+vel2+vK).M()> MaxMB_Cut || (vel1+vel2+vK).M()< MinMB_Cut) continue;
+         if ((vel1+vel2+vK).Pt()<PtB_Cut) continue;
+         auto trantrk=std::make_shared<reco::TransientTrack>(reco::TransientTrack(*trk,&(*bFieldHandle)));
+	 tempTracks.clear();
```

Comment:
better to declare tempTracks here - then you can also give it the right size

Reply:
Ok done